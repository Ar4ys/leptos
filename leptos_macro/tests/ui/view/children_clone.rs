use leptos::*;

#[slot]
struct Then {
    children: Children,
}

#[component]
fn Component(children: Children) -> impl IntoView {
    _ = children;
}

#[component]
fn Slot(then: Then) -> impl IntoView {
    _ = then;
}

struct A;

fn main() {
    let a = A;

    view! {
        <Component clone:a>
            <p />
        </Component>
    };

    view! {
        <Slot>
            <Then slot clone:a>
                <p/>
            </Then>
        </Slot>
    };
}
