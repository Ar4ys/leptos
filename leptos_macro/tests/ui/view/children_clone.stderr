error[E0277]: the trait bound `A: Clone` is not satisfied
  --> tests/ui/view/children_clone.rs:24:20
   |
23 | /     view! {
24 | |         <Component clone:a>
   | |                    ^^^^^^^ the trait `Clone` is not implemented for `A`
25 | |             <p />
26 | |         </Component>
27 | |     };
   | |_____- required by a bound introduced by this call
   |
help: consider annotating `A` with `#[derive(Clone)]`
   |
18 + #[derive(Clone)]
19 | struct A;
   |

error[E0277]: the trait bound `A: Clone` is not satisfied
  --> tests/ui/view/children_clone.rs:31:24
   |
29 | /     view! {
30 | |         <Slot>
31 | |             <Then slot clone:a>
   | |                        ^^^^^^^ the trait `Clone` is not implemented for `A`
32 | |                 <p/>
33 | |             </Then>
34 | |         </Slot>
35 | |     };
   | |_____- required by a bound introduced by this call
   |
help: consider annotating `A` with `#[derive(Clone)]`
   |
18 + #[derive(Clone)]
19 | struct A;
   |
